Wikipedia/Code/WMFAuthLinkLabel.swift

```diff
@@ -7,8 +15,27 @@ class WMFAuthLinkLabel: UILabel {
         lineBreakMode = .byWordWrapping
         textAlignment = .natural
     }
+
+    /// Some auth labels display a string from two localized strings, each styled differently.
+    public var strings: WMFAuthLinkLabelStrings?
+
     override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
         super.traitCollectionDidChange(previousTraitCollection)
-        font = UIFont.wmf_preferredFontForFontFamily(WMFFontFamily.systemBold, withTextStyle: .subheadline, compatibleWithTraitCollection: self.traitCollection)
+        
+        let boldSubheadlineFont = UIFont.wmf_preferredFontForFontFamily(WMFFontFamily.systemBold, withTextStyle: .subheadline, compatibleWithTraitCollection: self.traitCollection)!
+
+        guard let strings = strings else {
+            // Not combining two strings.
+            font = boldSubheadlineFont
+            return
+        }
+
+        // Combine and style 'dollarSignString' and 'substitutionString'.
+        // https://github.com/wikimedia/wikipedia-ios/pull/1216#discussion_r104224511
+        let subheadlineFont = UIFont.wmf_preferredFontForFontFamily(WMFFontFamily.system, withTextStyle: .subheadline, compatibleWithTraitCollection: self.traitCollection)!
```

Comment:
@joewalsh do you think we can make "wmf_preferredFontForFontFamily" return font instead of an optional font?

I'll fix the "!" here, but was just curious.